src/metabase/models/card.clj

```diff
@@ -46,6 +47,13 @@
       first
       :dashboards))
 
+(defn url
+  "Return a URL that can be used to access this Card."
+  [{:keys [id]}]
+  (when id
+    (str (setting/get :-site-url) "/card/" id)))
```

Comment:
We have a util class with functions for forming urls to various things ... https://github.com/metabase/metabase/blob/master/src/metabase/util/urls.clj#L18-L23